ldso/dynlink.c

```diff
@@ -1763,14 +1767,14 @@ hidden void *__dls2(unsigned char *base, size_t *sp)
 
 void *__dls2b(size_t *sp)
 {
-//	/* Setup early thread pointer in builtin_tls for ldso/libc itself to
-//	 * use during dynamic linking. If possible it will also serve as the
-//	 * thread pointer at runtime. */
-//	libc.tls_size = sizeof builtin_tls;
-//	libc.tls_align = tls_align;
-//	if (__init_tp(__copy_tls((void *)builtin_tls)) < 0) {
-//		a_crash();
-//	}
+	/* Setup early thread pointer in builtin_tls for ldso/libc itself to
+	 * use during dynamic linking. If possible it will also serve as the
+	 * thread pointer at runtime. */
+	// libc.tls_size = sizeof builtin_tls;
+	// libc.tls_align = tls_align;
+	// if (__init_tp(__copy_tls((void *)builtin_tls)) < 0) {
+	// 	a_crash();
+	// }
```

Comment:
Is this meant to still be commented out?

Reply:
This can be uncommented. The assumption of the thread pointer being used at runtime however, doesn't hold for the current setup. The thread which calls __dls2b is the initial ethread, but the thread which calls __dls3(which sets the tls for application proper) is already an lthread.

So if